src/payments/views.py

```diff
@@ -0,0 +1,199 @@
+from .models import Payments
+from rest_framework import filters
+from django.conf import settings
+from django.core.paginator import PageNotAnInteger, EmptyPage, Paginator
+
+
+class Customer(APIView):
+
+    def post(self, request):
+        serializer = CustomerSerializer(data=request.data)
+        if serializer.is_valid(raise_exception=True):
+            try:
+                customer = serializer.create(user=request.user)
+                data = {
+                    "email": customer.email,
+                    'name': customer.name,
+                    'phone': customer.phone,
+                    'description': customer.description,
+                    'currency': customer.currency,
+                    'balance': customer.balance,
+                    'created': customer.created
+                }
+                return Response(data=data, status=status.HTTP_201_CREATED)
+            except Exception as e:
+                return Response({'error_msg': str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class PaymentMethod(APIView):
+
+    def get(self, request):
+        try:
+            list_payment_method = PaymentService.get_list_payments_method_custommer(user=request.user)
+            context= []
+            for item in list_payment_method.data:
+                data = {
+                    'id': item.id,
+                    'brand': item.card.brand,
+                    'country': item.card.country,
+                    'expired_month': item.card.exp_month,
+                    'expired_year': item.card.exp_year,
+                    'last4': item.card.last4,
+                    'created': item.created
+                }
+                context.append(data)
+
+            return Response(data=context, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({'error_msg': str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+    def post(self, request):
+        serializer = PaymentMethodSerializer(data=request.data)
+        if serializer.is_valid(raise_exception=True):
+            try:
+                payment_method = serializer.create(user=request.user)
+                data = {
+                    'id': payment_method.id,
+                    'brand': payment_method.card.brand,
+                    'country': payment_method.card.country,
+                    'expired_month': payment_method.card.exp_month,
+                    'expired_year': payment_method.card.exp_year,
+                    'last4': payment_method.card.last4,
+                    'created': payment_method.created
+                }
+                return Response(data=data, status=status.HTTP_201_CREATED)
+            except Exception as e:
+                return Response({'error_msg': str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class PaymentIntent(APIView):
+
+    def post(self, request):
+        tour_id = request.query_params.get('tour_id')
+        tour = Tours.objects.get(id=tour_id)
```

Comment:
It `Tours.objects.get(id=tour_id)` raise exception when have not found any tour object?